File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/user/exceptions/FailRenewException.java
package com.hgianastasio.biblioulbrav2.core.user.exceptions;

/**
 * Created by heitorgianastasio on 4/26/17.
 */

public class FailRenewException extends Exception {
    String cause;

    public FailRenewException(String cause) {
        this.cause = cause;
    }
    public FailRenewException(String cause, Throwable throwable) {
        super(throwable);
        this.cause = cause;
    }

    @Override
    public String getMessage() {
        return String.format("Não foi possível renovar os empréstimos.\nRazão: %s",cause);
    }
}

File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/user/exceptions/FailRenewException.kt
package com.hgianastasio.biblioulbrav2.core.user.exceptions

/**
 * Created by heitorgianastasio on 4/26/17.
 */
class FailRenewException : Exception {
    var reason: String?

    constructor(cause: String?) {
        this.reason = cause
    }

    constructor(cause: String?, throwable: Throwable?) : super(throwable) {
        this.reason = cause
    }

    override val message: String
        get() = String.format("Não foi possível renovar os empréstimos.\nRazão: %s", reason)
}